backend/src/sms/middlewares/sms-transactional.middleware.ts

```diff
@@ -0,0 +1,56 @@
+import type { Request, Response, NextFunction } from 'express'
+import { SmsTransactionalService } from '@sms/services'
+import { loggerWithLabel } from '@core/logger'
+import { TemplateError } from 'postman-templating'
+import { RecipientError, RateLimitError } from '@core/errors'
+
+const logger = loggerWithLabel(module)
+
+async function sendMessage(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  const ACTION = 'sendMessage'
+
+  const { credentials } = res.locals
+  try {
+    const { recipient, body } = req.body
+
+    logger.info({ message: 'Sending SMS', action: ACTION })
+    await SmsTransactionalService.sendMessage({
+      credentials,
+      recipient,
+      body,
+    })
+    res.sendStatus(202)
+  } catch (err) {
+    logger.error({
+      message: 'Failed to send SMS',
+      error: err,
+      action: ACTION,
+    })
+
+    const BAD_REQUEST_ERRORS = [TemplateError, RecipientError]
+    if (BAD_REQUEST_ERRORS.some((errType) => err instanceof errType)) {
+      res.status(400).json({ message: err.message })
+      return
+    }
+
+    if (err instanceof RateLimitError) {
+      logger.warn({
+        message: 'Rate limited request to send transactional SMS',
+        userId: req?.session?.user.id,
+        label: credentials.label,
```

Comment:
I've decided to log just the `userId` and `label` here, since the rest of the credentials can be determined from just these 2 if needed. Also, I was a bit unsure about logging the credentials directly since they reveal somewhat sensitive information

Reply:
I think User ID and label are good enough!